discover.go

```diff
@@ -59,6 +60,7 @@ var Providers = map[string]Provider{
 	"triton":       &triton.Provider{},
 	"vsphere":      &vsphere.Provider{},
 	"packet":       &packet.Provider{},
+	"hcloud":       &hcloud.Provider{},
```

Comment:
Question for @dnephin or @alvin-huang:

What should the determining factors be to have a provider added to the default provider list? If `k8s` isn't a default should this new provider be part of the default set? Or maybe `k8s` should be in the default too?

Reply:
I think with the switch to Go modules we lost the ability to pull in go-discover without k8s dependencies so maybe we should just put k8s in the default set (or separate out that provider dir into its own go module.

Reply:
I think we probably forgot to add `k8s` into the map at the time? I don't recall a particular reason why it was excluded.

Reply:
Actually there's some notes in the original PR https://github.com/hashicorp/go-discover/pull/55

Reply:
Yeah at the time we could have per-provider dependencies that wouldn't get used unless you pulled in that package. With Go modules that is no longer the case unless we were to add a go.mod to the k8s provider and version it specifically.

For this PR I see no reason to not make it default if the only reason k8s wasn't was due to dependencies.